python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -976,9 +976,9 @@ def _target(self) -> Optional[Union[DirectTarget, RepoRelativeTarget]]:
     @property
     def job_name(self) -> Optional[str]:
         """Optional[str]: The name of the job that is targeted by this sensor."""
-        if len(self._targets) > 1:
+        if len(self._targets) != 1:
```

Comment:
Found this would throw `IndexError` for the `0` case, makes more sense for it to throw the same error as for multiple targets.